contracts/operators/Beefy/BeefyVaultOperator.sol

```diff
@@ -25,12 +33,14 @@ contract BeefyVaultOperator {
     /// @return tokens Array of token addresses
     ///         - [0] : The vault token received address
     ///         - [1] : The token deposited address
-    function deposit(uint256 amount, uint256 minVaultAmount)
-        external
-        payable
-        returns (uint256[] memory amounts, address[] memory tokens)
-    {
+    function deposit(
+        IERC20 vault,
```

Comment:
💅 `vault`  type should be of type address here:
1. it matches the natspec description
2. it is easier to understand
3. you don't have to cast it to address L42
4. For the sake of consistency, it is passed as addresses in the constructor already